### PR TITLE
Merge Lab 4

### LINKED ISSUE #4: PC design

### LINKED ISSUE #6: Complete REGFILE & ALU

### LINKED ISSUE #7: Complete Testbench and Verification

### LINKED ISSUE #8: Complete Control Unit

### PR DESCRIPTION
**Pull request by EZePiZy**

### CODE REVIEW

**Comment by Diegovano** [comment]
Please rename the top level folder to PC (as per the filename convention) the idea is that top level file names match the name of their respective component.

File: PC/Mux.sv
Comment (Diegovano) [inline PC/Mux.sv]: Not sure a separate mux module is needed, but I agree that in this case it makes the operation very clear.

**Review by Diegovano — COMMENTED** [review]
Luigi was saying that the instruction memory doesn't work yet. Is that still the case or can we merge?

**Comment by Diegovano** [comment]
Your README is excellent @EZePiZy, and I will be inspiring myself with it!

**Review by luigirinaldi — APPROVED** [review]
Nice stuff

**Comment by Diegovano** [comment]
Luigi doesn't agree on the file organisation so we will be reviewing it on Thursday. Until then, please do not merge with `main`. Thanks

**Review by Diegovano — APPROVED** [review]
Ready for merging!

**Pull request merged**
